shared/idmap/shift_linux.go

```diff
@@ -18,11 +19,33 @@ import (
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/capability.h>
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/acl.h>
 
+// Needs to be included at the end
+#include <attr/xattr.h>
+
+int set_caps(char *path, char *caps, ssize_t len, uint32_t uid) {
+	ssize_t ret;
+	struct vfs_ns_cap_data ns_xattr;
+
+	memset(&ns_xattr, 0, sizeof(ns_xattr));
+	memcpy(&ns_xattr, caps, len);
+	ns_xattr.magic_etc &= ~(VFS_CAP_REVISION_1 | VFS_CAP_REVISION_2);
+	ns_xattr.magic_etc |= VFS_CAP_REVISION_3 | VFS_CAP_FLAGS_EFFECTIVE;
```

Comment:
That needs to be:
```
ns_xattr.magic_etc |= VFS_CAP_REVISION_3;
```

`VFS_CAP_FLAGS_EFFECTIVE` only needs to be set if it is already set. :)